base/internal/message.hpp

```diff
@@ -43,6 +50,20 @@ template <class Key, class T, class Hash = std::hash<Key>, class Pred = std::equ
 inline std::string DebugPrint(std::unordered_map<Key, T, Hash, Pred> const & v);
 //@}
 
+#if defined has_cpp20_deleted_overloads
+template <>
+struct std::formatter<char16_t, char>: std::formatter<char> {
+  auto format(const char16_t& ch16, std::format_context& ctx) const {
+    return std::formatter<char>::format(ch16, ctx);
+  }
+};
+
+inline std::string DebugPrint(char16_t t)
+{
+  return std::format("{}", t);
```

Comment:
Note that `std::format` requires using XCode 15.3, we won't be able to build and test for iOS 12, 13, and 14 simulators after enabling it because it is available only in XCode 14.

Reply:
This builds just fine with any older compiler which was able to build the code before because this is inside of `#if defined has_cpp20_deleted_overloads`.